lib/puppet/provider/templates/webadministration/logproperties.ps1.erb

```diff
@@ -1,9 +1,44 @@
+$iis_version = '<%= Facter.value(:iis_version) %>'
+
+$W3CValue = 'W3C'
+$MaxSizeValue = 'MaxSize'
+
+# Convert the string representation of an Enum value into an Int32 safely
+Function ConvertStringTo-EnumInt32($EnumType, $Value, $ValueName) {
```

Comment:
Updated. I've left it as EnumType as it's an Object of type System.Type.  I would rather not use the var name `$Enum` as it can get confusing with the keyword Enum and the Type System.Enum (`[Enum]`)